### DISCUSSION TITLE
Migrating to Akka.Persistence.Sql: when can I disable deleteCompatibilityMode?

### DISCUSSION THREAD

**Discussion by robindv** [akkadotnet/akka.net#7389] [Q&A] [answered]
Hi,

I'm migrating an existing Akka.Persistence.PostgreSql-project to Akka.Persistence.Sql. The youtube-video and the migration-guide are extremely helpful, so its working like a charm.

The backwards-compatibility consists of three settings: tagStorageMode, useWriterUuidColumn and the deleteCompatibilityMode.
The first two are completely covered, but when you should be able to set the deleteCompatibilityMode to false stays unclear to me.

In my project I didn't do any deletes, so the metadata-table is empty and all events in the journal have is_deleted=false. Based on experiments switching the setting and looking at the source code of Akka.Persistence.Sql my conclusion would be that in my case I can just flip it to false and remove the obsolete metadata-table. This would probably also come with an performance increase since it needs to do fewer lookups in an empty table.

But since this is undocumented, I would like some confirmation. And perhaps its something that could be added to the docs.

Thanks!

Robin

**Comment by Aaronontheweb** [comment]
> But since this is undocumented, I would like some confirmation. And perhaps its something that could be added to the docs.

That's a great suggestion - we'll do that. I'll confer with @Arkatufus during our morning meeting today and we'll prepare a canonical answer for you here.

**Comment by Arkatufus** [comment, marked as answer]
Yes, **if** the journal metadata table is completely empty **and** none of the journal entries were marked as deleted, it is safe to set `deleteCompatibilityMode` to false and to delete the journal metadata table.

**Reply by robindv** [reply]
Clear, thanks for the confirmation!
